lib/standard/fixints.nit

```diff
@@ -0,0 +1,976 @@
+	private fun native_to_s(nstr: NativeString, strlen: Int) `{
+		snprintf(nstr, strlen, "%hhd", self);
+	`}
+
+	# Displayable Int8
+	#
+	#     assert 1i8.to_s       == "1"
+	#     assert (-123i8).to_s  == "-123"
+	redef fun to_s do
+		var nslen = to_s_len
+		var ns = new NativeString(nslen + 1)
+		ns[nslen] = 0u8
+		native_to_s(ns, nslen + 1)
+		return ns.to_s_with_length(nslen)
+	end
+end
+
+# Native 16-bit signed integer.
+# Same as a C `int16_t`
+universal Int16
+	super Discrete
+	super Numeric
+
+	redef type OTHER: Int16
+
+	redef fun successor(i) do return self + i.to_i16
+	redef fun predecessor(i) do return self - i.to_i16
+
+	redef fun object_id is intern
+	redef fun hash do return self.to_i
+	redef fun ==(i) is intern
+	redef fun !=(i) is intern
+	redef fun output is intern
+
+	redef fun <=(i) is intern
+	redef fun <(i) is intern
+	redef fun >=(i) is intern
+	redef fun >(i) is intern
+	redef fun +(i) is intern
+
+	#     assert -1i16 == 0xFFFFi16
+	#     assert -0i16 == 0i16
+	redef fun - is intern
+	redef fun -(i) is intern
+	redef fun *(i) is intern
+	redef fun /(i) is intern
+
+	# Modulo of `self` with `i`.
+	#
+	# Finds the remainder of division of `self` by `i`.
+	#
+	#     assert 5i16 % 2i16	== 1i16
+	#     assert 10i16 % 2i16	== 0i16
+	fun %(i: Int16): Int16 is intern
+
+	redef fun zero do return 0.to_i16
+	redef fun value_of(val) do return val.to_i16
+
+	# `i` bits shift fo the left
+	#
+	#     assert 5i16 << 1 == 10i16
+	fun <<(i: Int): Int16 is intern
+
+	# `i` bits shift fo the right
+	#
+	#     assert 5i16 >> 1 == 2i16
+	fun >>(i: Int): Int16 is intern
+
+	redef fun to_i is intern
+	redef fun to_f is intern
+	redef fun to_b is intern
+	redef fun to_i8 is intern
+	redef fun to_i16 do return self
+	redef fun to_u16 is intern
+	redef fun to_i32 is intern
+	redef fun to_u32 is intern
+
+	redef fun distance(i) do return (self - i).to_i
+
+	redef fun <=>(other)
+	do
+		if self < other then
+			return -1
+		else if other < self then
+			return 1
+		else
+			return 0
+		end
+	end
+
+	redef fun is_between(c, d)
+	do
+		if self < c or d < self then
+			return false
+		else
+			return true
+		end
+	end
+
+	redef fun max(other)
+	do
+		if self < other then
+			return other
+		else
+			return self
+		end
+	end
+
+	redef fun min(c)
+	do
+		if c < self then
+			return c
+		else
+			return self
+		end
+	end
+
+	# Returns the result of a binary AND operation on `self` and `i`
+	#
+	#     assert 0x10i16 & 0x01i16 == 0i16
+	fun &(i: Int16): Int16 `{ return self & i; `}
+
+	# Returns the result of a binary OR operation on `self` and `i`
+	#
+	#     assert 0x10i16 | 0x01i16 == 0x11i16
+	fun |(i: Int16): Int16 `{ return self | i; `}
+
+	# Returns the result of a binary XOR operation on `self` and `i`
+	#
+	#     assert 0x101i16 ^ 0x110i16 == 0x11i16
+	fun ^(i: Int16): Int16 `{ return self ^ i; `}
+
+	# Returns the 1's complement of `self`
+	#
+	#     assert ~0x2Fi16 == 0xFFD0i16
+	fun ~: Int16 `{ return ~self; `}
```

Comment:
But waht about the 2's complement? Shouldn't the `~` method take a param to select that base?